### PR TITLE
Default Embedded format

### PR DESCRIPTION
**Pull request by habdelra**
This PR introduces a default embedded view that is capable of rendering differently based on the container size that is yielded to it. Note that the large thumbnail design has some fundamental issues regarding the integrity of the card boundary, so i eschewed the current design and did my best to make something tasteful at the large thumbnail breakpoint.

image

This is what it looks like for cards without a thumbnail image:

image

### CODE REVIEW

File: packages/base/card-api.gts [hunk 1]
```diff
@@ -1783,6 +1786,188 @@ class DefaultCardDefTemplate extends GlimmerComponent<{
...
+              <div class='thumbnail-subsection'>
+                <h4
+                  class='card-display-name'
+                  data-test-card-display-name
+                >{{cardTypeDisplayName @model}}</h4>
+              </div>
+            </div>
+          {{else}}
+            {{! empty links-to field }}
+            <div data-test-empty-field class='empty-field'></div>
```
Comment (habdelra) [inline packages/base/card-api.gts]: probably we should show something reasonable here. it seems this was always rendering for empty `<@field.emptyLinkValue/>`.

**Comment by habdelra** [comment]
we still have issues like this that we need to sort thru where in the places that the embedded cards are being rendered we are not catering for the new design. Note that the smallest embedded card format is a bit too large to fit in the form's grid, as well as the border for the embedded card clashes with the form

image

**Comment by habdelra** [comment]
The percy diffs are worth noting. I'm gonna just leave them there so people can see how this have changed. There are a few diffs with missing stuff that I'm still working thru. One of the main things tho are some card boundary integrity issues--specifically padding and borders. that used to be drawn outside of the card for the embedded view. it might make more sense to now handle that inside of the card for embedded view, otherwise we could still draw this outside of the cards, but we'll need to do some style math to reverse engineer the breakpoints. the percy diffs illustrate how for the non-default embedded cards, they were relying on padding and borders being provided by the system that now are up to the embedded template in the current form of this PR.

File: packages/base/field-component.gts [hunk 2]
```diff
@@ -244,7 +244,18 @@ export function getBoxComponent(
...
+          in order for the container queries to make sense--otherwise we need to do style 
+          math to figure out what the actual breakpoints are. please resist the urge to add 
+          padding, borders, etc--bascially anything that alters the geometry inside of the 
+          card boundary.
+
+          we need to use height 100% because the container query for embedded cards only
+          works if we use up all the space horizontally and vertically that is available
+          to the card since some of our queries are height queries
+        */
+        height: 100%;
```
Comment (tintinthong) [inline packages/base/field-component.gts]: 👍

**Review by jurgenwerk — APPROVED** [review]
Nice!

**Review by tintinthong — APPROVED** [review]

**Comment by habdelra** [comment]
From the boxel sprint planning: 
cards do not draw their own borders, however cards do own the padding within the border

File: packages/base/card-api.gts [hunk 3]
```diff
@@ -1783,6 +1786,188 @@ class DefaultCardDefTemplate extends GlimmerComponent<{
...
+    </div>
+    <style>
+      .embedded-template {
+        width: 100%;
+        height: 100%;
+        display: flex;
+      }
+      .container {
+        container-name: embedded-card;
+        container-type: size;
```
Comment (habdelra) [inline packages/base/card-api.gts]: we should move this CSS container into the field component based on the discussion from the sprint review

File: packages/base/card-api.gts [hunk 4]
```diff
@@ -1783,6 +1786,188 @@ class DefaultCardDefTemplate extends GlimmerComponent<{
...
+        width: 100%;
+      }
+      .contents {
+        height: 100%;
+        padding: 10px;
+        display: flex;
+        flex-wrap: wrap;
+        padding: 11px;
+        border: var(--boxel-border-card);
+        border-radius: var(--boxel-border-radius-lg);
```
Comment (burieberry) [inline packages/base/card-api.gts]: My understanding from sprint planning discussion is that the border styles will need to move outside the card as well.

File: packages/base/card-api.gts [hunk 5]
```diff
@@ -1783,6 +1786,188 @@ class DefaultCardDefTemplate extends GlimmerComponent<{
...
+        container-name: embedded-card;
+        container-type: size;
+        width: 100%;
+      }
+      .contents {
+        height: 100%;
+        padding: 10px;
+        display: flex;
+        flex-wrap: wrap;
+        padding: 11px;
```
Comment (burieberry) [inline packages/base/card-api.gts]: padding is stated twice here

**Review by burieberry — APPROVED** [review]

File: packages/base/card-api.gts (hunk 4 above)
Comment (habdelra) [reply packages/base/card-api.gts]: yeah, i noted that above

**Comment by habdelra** [comment]
I took executive privilege regarding the design docs, and adjusted the "strip" style card layout to work with the edit format's form. this meant reducing the height of the strip style to 40px. We'll need to address this in the design doc

Screenshot from 2024-07-08 11-26-53

**Comment by habdelra** [comment]
For these types of percy issues with the missing padding, instead of introducing padding into all the embedded templates (since the cards own the style within the card boundary), let's hold off until we have a way to request the prerendered CardDef embedded HTML to insert into the area yielded to the field. I think we'll want to use the CardDef embedded fields for edit mode. That will probably be the correct way to handle this. That should be a few PR's from now.

image

**Comment by habdelra** [comment]
There are also percy diffs that look like this that appear to have missing content. I double checked, in actuality, the content is there, it's just that it is being rendered off the screen. this is more fallout from injecting styles (e.g. padding) within the card boundary. The solution here is that cards own the entire space yielded to them, from edge to edge. if they are styled wonky, they will look wonky. we should not be relying on container provided CSS within a card boundary for the delegated render.

image

**Review by backspace — APPROVED** [review]
Thanks for all the context

**Comment by lukemelia** [comment]
>  I think we'll want to use the CardDef embedded fields for edit mode.

No need to change anything in this PR, but I'll note that I don't think this will be the case. I think in the case of a LinksTo field with the parent card in edit mode, we would want to use the Card's own embedded format, if it has one.

**Pull request merged**